crates/subspace-farmer/src/multi_farming.rs

```diff
@@ -179,6 +182,33 @@ impl MultiFarming {
             })
             .await?;
 
+            // Enable DSN subscription for segments archiving.
+            if enable_dsn_archiving {
+                tokio::spawn({
+                    let node = node.clone();
+                    async move {
+                        trace!("Subscribing to pubsub archiving...");
+                        match node
+                            .subscribe(subspace_networking::PUB_SUB_ARCHIVING_TOPIC.clone())
+                            .await
+                        {
+                            Ok(mut subscription) => {
+                                info!("Subscribed to pubsub archiving.");
+
+                                //TODO: Integrate with archiving process.
+                                #[allow(clippy::redundant_pattern_matching)]
```

Comment:
Why does it complain here though?

Reply:
I didn't use the result from the `subscription.next()`. But it's a place holder for the future integration.